edx_solutions_api_integration/management/commands/send_progress_leaderboard_notifications.py

```diff
@@ -47,16 +49,15 @@ def handle(self, *args, **options):
             aggregation_name='course',
             last_modified__gte=time_range
         ).distinct().values_list('course_key', flat=True)
-
         for course_key in courses:
             all_progress = Aggregator.objects.filter(
-                aggregation_name='course', course_key=course_key, percent__gt=0
+                aggregation_name='course', course_key=course_key, percent__gt=0, user__is_staff=False
+            ).exclude(user__groups__groupprofile__name__in=['mcka_role_mcka_observer', 'mcka_role_mcka_ta']
```

Comment:
I think you should check against `student_courseaccessrole` instead(also course key should be in query otherwise it will filter users having staff/ta role from other courses too),
Note: user.is_staff and staff role against a course are two different things.

Reply:
thanks for the tip about "staff"  :)